psi_cardinality/go/client/client.go

```diff
@@ -0,0 +1,85 @@
+package client
+
+/*
+#include "psi_cardinality/c/psi_cardinality_client_c.h"
+*/
+import "C"
+import (
+	"errors"
+	"fmt"
+	"runtime"
+)
+
+//PSICardinalityClient context
+type PSICardinalityClient struct {
+	context C.psi_cardinality_client_ctx
+}
+
+//Create returns a new PSI client
+func Create() (*PSICardinalityClient, error) {
+	psiClient := new(PSICardinalityClient)
+
+	var err *C.char
+	ret := C.psi_cardinality_client_create(&psiClient.context, &err)
+	if ret != 0 {
+		return nil, fmt.Errorf("failed to create client context: %v(%v)", C.GoString(err), ret)
+	}
+	if psiClient.context == nil {
+		return nil, errors.New("failed to create client context: null")
```

Comment:
Can this ever happen if there isn't an implementation error in our library? If not, maybe make this explicit, e.g. `errors.New("failed to create client context: Context is NULL. This should never happen")`